src/basic.jl

```diff
@@ -142,3 +152,19 @@ function reweight(m, b, args...)
     return wm
 end
 reweight(f::BasicParticleFilter, args...) = reweight(f.reweight_model, args...)
+
+
+"""
+This function computes Effective Sample Size (ESS) which estimates the number of particles that are effectively contributing to the approximation of the target distribution. 
+
+ESS is divided by `num_particles` to make it easier to get a uniform threshold (scale between 0 and 1) for resampling across different particle filters.
+
+M. S. Arulampalam, S. Maskell, N. Gordon and T. Clapp, "A tutorial on particle filters for online nonlinear/non-Gaussian Bayesian tracking," in IEEE Transactions on Signal Processing, vol. 50, no. 2, pp. 174-188, Feb. 2002, doi: 10.1109/78.978374.
+keywords: {Tutorial;Particle filters;Nonlinear dynamical systems;Costs;Signal processing;Bayesian methods;Particle tracking;Kalman filters;Filtering;Monte Carlo methods},
+"""
+function calculate_ess(weights)
```

Comment:
```suggestion
function normalized_ess(weights)
```
"calculate" does not really add anything to the name of this function. Also, If someone just said "ESS", does it mean a number between 0 and 1 or an actual number of particles? If ESS means a number of particles, I think we should call this function `normalized_ess`.

Reply:
changed the name to `normalized_ess`